src/main/java/io/debezium/connector/cassandra/KafkaRecordEmitter.java

```diff
@@ -41,28 +43,35 @@
 
     public KafkaRecordEmitter(String kafkaTopicPrefix, String heartbeatPrefix, Properties kafkaProperties,
                               OffsetWriter offsetWriter, Duration offsetFlushIntervalMs, long maxOffsetFlushSize,
-                              Converter keyConverter, Converter valueConverter) {
+                              Converter keyConverter, Converter valueConverter, HashSet<String> errorCommitLogSet,
+                              CommitLogTransfer commitLogTransfer) {
         this.producer = new KafkaProducer<>(kafkaProperties);
         this.topicSelector = CassandraTopicSelector.defaultSelector(kafkaTopicPrefix, heartbeatPrefix);
         this.offsetWriter = offsetWriter;
         this.offsetFlushPolicy = offsetFlushIntervalMs.isZero() ? OffsetFlushPolicy.always() : OffsetFlushPolicy.periodic(offsetFlushIntervalMs, maxOffsetFlushSize);
+        this.errorCommitLogSet = errorCommitLogSet;
+        this.commitLogTransfer = commitLogTransfer;
         this.keyConverter = keyConverter;
         this.valueConverter = valueConverter;
     }
 
     public void emit(Record record) {
         try {
             synchronized (lock) {
-                ProducerRecord<byte[], byte[]> producerRecord = toProducerRecord(record);
                 LOGGER.debug("Sending the record '{}'", record.toString());
+                ProducerRecord<byte[], byte[]> producerRecord = toProducerRecord(record);
                 Future<RecordMetadata> future = producer.send(producerRecord);
                 LOGGER.debug("The record '{}' has been sent", record.toString());
                 futures.put(record, future);
                 maybeFlushAndMarkOffset();
             }
         }
         catch (Exception e) {
-            throw new DebeziumException(String.format("Failed to send record %s", record.toString()), e);
+            LOGGER.error("Failed to send the record {}. Error: ", record.toString(), e);
```

Comment:
I'd suggest to _either_ log _or_ rethrow, so not to have the same exception in the logs twice.

Reply:
> I'd suggest to _either_ log _or_ rethrow, so not to have the same exception in the logs twice.

Updated as below so that we won't have duplicate logs:
- When we need to throw the error, we throw DebeziumException.
- When we need to catch the error, we log the error message.